Src/SwqlStudio/LexerService.cs

```diff
@@ -28,7 +28,13 @@ private class SwisEntity
         }
 
         private readonly ILexerDataSource _lexerDataSource;
-        private static List<string> BasicAutoCompletionKeywords { get; }
+
+        public static readonly Tuple<int, IEnumerable<string>>[] LexerKeywords =
+        {
+            new Tuple<int, IEnumerable<string>>(0, Grammar.General),
+            new Tuple<int, IEnumerable<string>>(4, Grammar.Functions),
+            new Tuple<int, IEnumerable<string>>(5, Grammar.AggregateFunctions)
```

Comment:
Minor: I realize the 0, 4, 5 constants already existed in the original code, but it might be helpful to introduce named constants to clarify their purpose.  Introducing the `Grammar` class definitely helps, though.

Reply:
Sure, i only did the reafactoring, i dont know what the constants meain the User control library.